api/v1alpha1/kubeadmbootstrapconfig_types.go

```diff
@@ -18,21 +18,28 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeadmv1beta1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Phase defines KubeadmBootstrapConfig phases
+type Phase string
+
+// Ready defines the KubeadmBootstrapConfig Ready Phase
+const Ready Phase = "Ready"
 
 // KubeadmBootstrapConfigSpec defines the desired state of KubeadmBootstrapConfig
 type KubeadmBootstrapConfigSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	ClusterConfiguration kubeadmv1beta1.ClusterConfiguration `json:"clusterConfiguration"`
+	InitConfiguration    kubeadmv1beta1.InitConfiguration    `json:"initConfiguration"`
```

Comment:
Will a single config ever include both `InitConfiguration` and `JoinConfiguration`? If not, should we add the `omitempty` tag to these fields?

Reply:
As these objects are 1:1 to Machines, I believe it's either/or. 

+1 for `omitempty`. As it is now, I think the generator would mark these fields as required in the schema validation if my understanding of kubebuilder is correct.

Reply:
updated. Thanks!